core/src/test/java/greencity/controller/ShoppingListItemControllerTest.java

```diff
@@ -0,0 +1,177 @@
+
+    @Mock
+    private ShoppingListItemService shoppingListItemService;
+
+    @Mock
+    private Validator mockValidator;
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private ModelMapper modelMapper;
+
+    @InjectMocks
+    private ShoppingListItemController shoppingListItemController;
+    private static final String initialUrl = "/user/shopping-list-items";
+    private final Principal principal = getPrincipal();
+    private final UserVO userVO = getUserVO();
+
+    @BeforeEach
+    void setup() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(shoppingListItemController)
+                .setValidator(mockValidator)
+                .setCustomArgumentResolvers(new UserArgumentResolver(userService, modelMapper))
+                .build();
+    }
+
+    @Test
+    void bulkDeleteUserShoppingListItem_StatusOk() throws Exception {
+
+        when(userService.findByEmail(anyString())).thenReturn(userVO);
+
+        mockMvc.perform(delete(initialUrl)
+                        .param("habitId", "1")
+                        .param("shoppingListItemIds", "1,2")
+                        .principal(principal))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void saveUserShoppingListItemStatusWithoutLanguageParamTest_statusIsCreated() throws Exception {
+        List<UserShoppingListItemResponseDto> listItemResponseDtos = new ArrayList<>();
+
+        when(userService.findByEmail(anyString())).thenReturn(userVO);
+        when(shoppingListItemService.saveUserShoppingListItems(anyLong(), anyLong(), anyList(), anyString()))
+                .thenReturn(listItemResponseDtos);
+
+        mockMvc.perform(post(initialUrl)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("[]")
+                        .param("habitId", "1")
+                        .principal(principal))
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    void updateUserShoppingListItemStatusWithoutLanguageParamTest_statusIsCreated() throws Exception {
+        Long userShoppingListItemId = 1L;
+        String status = "DONE";
+
+        when(userService.findByEmail(anyString())).thenReturn(userVO);
+
+        mockMvc.perform(patch(initialUrl + "/{userShoppingListItemId}/status/{status}",
+                        userShoppingListItemId, status)
+                        .principal(principal))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void updateUserShoppingListItemStatusWithLanguageParamTest_statusIsCreated() throws Exception {
+        Long userShoppingListItemId = 1L;
+        String status = "DONE";
+        Locale locale = Locale.ENGLISH;
+
+        when(userService.findByEmail(anyString())).thenReturn(userVO);
+
+        mockMvc.perform(patch(initialUrl + "/{userShoppingListItemId}/status/{status}",
+                        userShoppingListItemId, status)
+                        .param("lang", locale.getLanguage())
```

Comment:
Probably you should use ".locale(locale)" instead of .param("lang", locale.getLanguage())
I suspect that ".param("lang", locale.getLanguage())" does not work in this case.

Reply:
Thanks, fixes are done